### PR TITLE
Adds GrpcCurrentTraceContext

### PR DESCRIPTION
**Pull request by adriancole**
See #425

### CODE REVIEW

File: context/grpc/src/main/java/brave/context/grpc/GrpcCurrentTraceContext.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package brave.context.grpc;
+
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.TraceContext;
+import io.grpc.Context;
+
+/** Manages scope with a {@link Context} as opposed to a thread local. */
+public final class GrpcCurrentTraceContext extends CurrentTraceContext {
```
Comment (adriancole) [inline context/grpc/src/main/java/brave/context/grpc/GrpcCurrentTraceContext.java]: @kuroneko25 can you copy/paste this file into your project, then wrap it with the log4j2 context and see if it works for you?

ex. `ThreadContextCurrentTraceContext.create(GrpcCurrentTraceContext.create())`

Comment (kuroneko25) [inline context/grpc/src/main/java/brave/context/grpc/GrpcCurrentTraceContext.java]: I wrapped the log4j2 context and that didn't work. But just to make sure I'm accessing it correctly `ThreadContext.get("traceId")` is the expected usage?

**Comment by adriancole** [comment]
@kuroneko25 so this had nothing to do with the bug, but I can merge if you are interested in using it instead of thread locals for other reasons. Eventually, we'd likely add something like this in order to join traces with instrumentation-java. Anyway, let me know.

**Comment by adriancole** [comment]
closing this as it was unrelated to the problem. If someone wants this we can re-open

**Pull request closed without merging**